packages/storage/src/persisted.ts

```diff
@@ -162,23 +179,88 @@ export function makePersisted<T, O extends Record<string, any> = {}>(
   if (init instanceof Promise) init.then(data => unchanged && data && set(data));
   else if (init) set(init);
 
+  if (typeof options.sync?.[0] === "function") {
+    options.sync[0]((data: PersistenceSyncData) => {
+      if (
+        data.key !== name ||
+        (!isServer && (data.url || globalThis.location.href) !== globalThis.location.href)
+      ) {
+        return;
+      }
+      set(data.newValue as string);
+    });
+  }
+
   return [
     signal[0],
     typeof signal[0] === "function"
       ? (value?: T | ((prev: T) => T)) => {
-          const output = (signal[1] as Setter<T>)(value as any);
-
-          if (value) storage.setItem(name, serialize(output), storageOptions);
-          else storage.removeItem(name, storageOptions);
-          unchanged = false;
-          return output;
-        }
+        const output = (signal[1] as Setter<T>)(value as any);
+        const serialized: string | null | undefined =
+          value != null ? (serialize(output) as string) : (value as null | undefined);
+        options.sync?.[1](name, serialized);
+        if (value != null) storage.setItem(name, serialized as string, storageOptions);
+        else storage.removeItem(name, storageOptions);
+        unchanged = false;
+        return output;
+      }
       : (...args: any[]) => {
-          (signal[1] as any)(...args);
-          const value = serialize(untrack(() => signal[0] as any));
-          // @ts-ignore
-          storage.setItem(name, value, storageOptions);
-          unchanged = false;
-        },
+        (signal[1] as any)(...args);
+        const value = serialize(untrack(() => signal[0] as any));
+        options.sync?.[1](name, value);
+        // @ts-ignore
+        storage.setItem(name, value, storageOptions);
+        unchanged = false;
+      },
   ] as typeof signal;
 }
+
```

Comment:
Btw. I am desperatly missing a Broadcast Sync, because as far as I know this is the official preferred way to communicate between tabs and windows

Reply:
Good point. I'll add it tomorrow, it's a bit late for me now.

Reply:
I changed postMessageSync to messageSync, so it can handle both post message and broadcast channel API.